src/words/ast/INodeMovesPredicate.java

```diff
@@ -10,7 +15,22 @@ public INodeMovesPredicate(Object... children) {
 
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		ASTValue className = children.get(0).eval(environment);
+		ASTValue objectAlias = children.get(1).eval(environment);
+		ASTValue moveDirection = children.get(2) == null ? null : children.get(2).eval(environment);
+		
+		HashSet<WordsObject> objectsOfClass = environment.getObjectsByClass(className.stringValue);
+
+		for (WordsObject object : objectsOfClass) {
```

Comment:
I had a similar concern, and I was going to bring this up tomorrow. I didn't want to add this functionality before we had listeners, but I think listeners might have to pass down their statement list in the eval function (using the new eval function from #87). Then, inside of this loop, we would create a local scope, run the statement list, and pop the local scope. 

This would result in the statement list being run each time that this predicate is true. I think this is easier than passing the result of the predicate (with scoping issues) up the tree. But I think this might be a good discussion item tomorrow.